blog/2023-07-12-angular-tutorial.md

```diff
@@ -0,0 +1,588 @@
+
+Use the command below to create a new Angular project. Make sure your environment is correctly set up to run [an angular project](https://angular.io/guide/setup-local).
+
+```bash
+ng new tolgee-ngx
+```
+
+You will be prompted to provide some specifications for your project.
+
+```
+? Would you like to add Angular routing? Yes
+? Which stylesheet format would you like to use? (Use arrow keys)
+> CSS
+  SCSS   [ <https://sass-lang.com/documentation/syntax#scss>                ]
+  Sass   [ <https://sass-lang.com/documentation/syntax#the-indented-syntax> ]
+  Less   [ <http://lesscss.org>                                             ]
+```
+
+```bash
+cd tolgee-ngx
+ng serve --open
+```
+
+This command launches your app, watches your files and rebuilds it whenever you make changes.
+The `--open` flag automatically opens your browsers to the `[localhost:4200](http://localhost:4200/)`.
+
+![Angular Application](/img/blog/angular-tutorial/angular_app.png)
+
+This is the basic application we will be translating with Tolgee. We will add a select component that will enable us to switch from one language to another.
+
+To translate your angular application we need to install the Tolgee package. Open another terminal window and run the command below in the project’s root directory.
+
+```bash
+npm i @tolgee/ngx
+```
+
+The command will install all the necessary dependencies.
+
+## Step 4: Add Environment Variables
+
+In the `src` directory create `environments/environment.local.ts` file and paste the following code:
+
+```typescript
+export const environment = {
+  production: false,
+  tolgeeApiUrl: 'Your server url here',
+  tolgeeApiKey: 'Your api key here',
+};
+```
+
+Replace `tolgeeApiUrl` with your server URL and `tolgeeApiKey` with the API key you generated earlier. Since this app is not for production, we set this variable to false.
+
+## Step 5: Setup Tolgee NgModule
+
+Open your `src/app/app.module.ts` and add the following code:
+
+```typescript
+...
+
+import {
+  DevTools,
+  NgxTolgeeModule,
+  Tolgee,
+  TOLGEE_INSTANCE,
+  FormatSimple,
+} from '@tolgee/ngx';
+import { environment } from 'src/environments/environment.local';
+
+...
+
+@NgModule({
+  ...
+  imports: [..., NgxTolgeeModule],
+
+  providers: [
+    {
+      provide: TOLGEE_INSTANCE,
+      useFactory: () => {
+        return Tolgee().use(DevTools()).use(FormatSimple()).init({
+          language: 'en',
+          apiUrl: environment.tolgeeApiUrl,
+          apiKey: environment.tolgeeApiKey,
+        });
+      },
+    },
+  ],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+Add the `NgxTolgeeModule` in the import section and provide a Tolgee instance in the providers. Our default application language is en you can change it to fr or es depending on your preferences. Lastly, we import the environment variable we added earlier which will help us to communicate with the server.
+
+## Step 6: Translate Your Texts and Export the files:
```

Comment:
From now on, it feels like you still need to get what's the most important benefit of Tolgee. With the toggle-SDK installed, you can add the keys by alt + clicking them. So you can add the key to the code and then translate it directly in the running app. You don't have to go to the platform and add it manualy. https://tolgee.io/js-sdk/in-context

I actually don't blame you for not getting it correctly. It's more that our documentation could be structured better so the reader who directly goes to the angular section finds out about this. Sorry! 😢